### ISSUE TITLE
namerd: should POST set ETag

### ISSUE THREAD

**Issue by olix0r** [BuoyantIO/linkerd#207]
```
> POST /api/1/dtabs/default HTTP/1.1
> Host: 104.197.46.186
> User-Agent: curl/7.43.0
> Accept: */*
> Content-type: application/dtab
> Content-Length: 192
> 
* upload completely sent off: 192 out of 192 bytes
< HTTP/1.1 204 No Content
< Content-Length: 0
< 
```

I'd expect creation to set an ETag on the repsonse?

**Comment by adleong** [comment]
in the case of zookeeper storage, the version number doesn't come back in the response from create.  that means that it would require a second call to zk to get the version number.  we can definitely do this, I just wanted to point out that it's not free.

**Comment by olix0r** [comment]
@adleong zookeeper itself gives us this for free, but we currently have some poor abstractions that hide it from us: The [`setData` api](https://zookeeper.apache.org/doc/r3.4.6/api/org/apache/zookeeper/ZooKeeper.html#setData(java.lang.String, byte[], int)) returns a [`Stat`](https://zookeeper.apache.org/doc/r3.4.6/api/org/apache/zookeeper/data/Stat.html#getMzxid()) that contains the modification mzxid.

I think we'll have to move away from the finagle-serverset abstractions...

**Comment by adleong** [comment]
I agree in general that we need to move away from the finagle-serverset abstractions to have the flexibility we want.  In this case, `setData` returns a `Stat` but `create` does not.

**Comment by adleong** [comment]
My last comment wasn't clear.  Zookeeper's `create` method does not return a `Stat` so I don't think moving off of finagle-serversets helps in this particular case.
